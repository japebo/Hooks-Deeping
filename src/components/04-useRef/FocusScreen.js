
import React, { useRef } from 'react'
import '../02-useEffect/effects.css'

export const FocusScreen = () => {
//este es un ejemplo de uso del hook useRef, pero es preferible hacerlo a travÃ©s de document.queryselector(), pues no hay que importar nada, y funciona de la misma forma. 
    const inputRef = useRef(); //crea una referencia mutable a cualquier elemento HTML, en este caso al elemento input, que se le quiere hacer un focus
    
    const handleClick = () => {
        // document.querySelector('input').select();
        inputRef.current.select();
    }

    return (
        <div>
            <h1>Focus Screen</h1>
            <hr/>
            <input 
                ref={ inputRef }
                className="form-control"
                placeholder="Su nombre"
            />
            <button 
                className="btn btn-outline-primary mt-3"
                onClick={ handleClick }
            >
                Focus
            </button>
        </div>
    )
}
