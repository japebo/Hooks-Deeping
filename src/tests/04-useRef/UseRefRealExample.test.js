import { shallow } from "enzyme"
import { UseRefRealExample } from "../../components/04-useRef/UseRefRealExample";

describe('Pruebas en <UseRefRealExample/>', () => {
    const wrapper = shallow(<UseRefRealExample/>);

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    })
    test('Debe mostrar el componente <UseRefRealExample/>', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })
    
})
