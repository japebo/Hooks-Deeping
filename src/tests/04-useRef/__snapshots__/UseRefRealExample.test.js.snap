// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <UseRefRealExample/> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    useRef Real Example
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
